actors/miner/src/expiration_queue.rs

```diff
@@ -388,8 +438,24 @@ impl<'db, BS: Blockstore> ExpirationQueue<'db, BS> {
 
                 recovered_power += &power;
             }
+            let mut faulty_unset = Vec::new();
+            for sector_number in expiration_set.faulty_sectors.iter() {
+                let sector = match remaining.remove(&sector_number) {
+                    Some(s) => s,
+                    None => continue,
+                };
+                // If the sector expires as faulty at this epoch, remove it for re-scheduling.
+                // It's not part of the on-time pledge number here.
+                faulty_unset.push(sector_number);
+                let power = power_for_sector(sector_size, sector);
+                faulty_power_delta -= &power;
+                sectors_rescheduled.push(sector);
+
+                recovered_power += &power;
+            }
+
             let mut early_unset = Vec::new();
-            for sector_number in expiration_set.early_sectors.iter() {
+            for sector_number in expiration_set.proof_expiring_sectors.iter() {
                 let sector = match remaining.remove(&sector_number) {
                     Some(s) => s,
                     None => continue,
```

Comment:
I don't think the following is strictly incorrect but unsetting and rescheduling these sectors is unnecessary.  Just like the on time sectors these proof_expiring_sectors are in their original expiration set, they were not rescheduled as faulty during faulting which is why they are tracked here instead of the faulty_sectors set.